script.js

```diff
@@ -1,4 +1,113 @@
+//1. Player 1 to roll dice
+//2. Player 1 to decide order of dice
+//3. Player 2 to roll dice
+//4. Player 2 to decide order of dice
+//5. Determine if Player 1 or 2 wins
+//6. Game continues with running score being tabulated
+
+var diceRoll_1 = "";
+var diceRoll_2 = "";
+var diceResult = "";
+var player = "Player 1";
+var winner = "";
+var gameMode = "start";
+var myOutputValue = "";
+var player1score = [];
+var player2score = [];
+var player1runningscore = "";
+var player2runningscore = "";
+var leaderboard = "";
+
+var diceRoll = function (input) {
+  if (input == "roll") {
+    diceRoll_1 = Math.floor(Math.random() * 6) + 1;
+    diceRoll_2 = Math.floor(Math.random() * 6) + 1;
+  }
+};
+
+var diceCombined = function (input) {
+  if (input == 1) {
+    diceResult = parseInt(diceRoll_1.toString() + diceRoll_2.toString());
+  } else if (input == 2) {
+    diceResult = parseInt(diceRoll_2.toString() + diceRoll_1.toString());
+  }
+};
+
+var runningScore = function (array) {
+  let sum = 0;
+
+  for (let i = 0; i < array.length; i++) {
+    sum += array[i];
+  }
+  return sum;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  while (gameMode == "start") {
+    //set current player
+    if (player1score.length > player2score.length) {
+      player = "Player 2";
+    } else {
+      player = "Player 1";
+    }
+    myOutputValue = `Welcome ${player}, enter "roll" to go.`;
+    gameMode = "play";
+    return myOutputValue;
+  }
+  //player to start rolling
+  while (gameMode == "play") {
+    if (input == "roll") {
+      diceRoll(input);
+      myOutputValue = `${player}, you have rolled ${diceRoll_1} and ${diceRoll_2}. Enter 1 or 2 to set the order`;
+    }
+    //player to choose order
+    if ((input == 1 || input == 2) && diceRoll_1 != "") {
+      diceCombined(input);
+
+      //record score
+      if (player == "Player 1") {
+        player1score.push(diceResult);
+        console.log(player1score);
+        myOutputValue = `${player}, your results are ${diceResult}. Next player, enter "roll" to go.`;
+        gameMode = "play";
+        player = "Player 2";
+      } else if (player == "Player 2") {
+        player2score.push(diceResult);
+        gameMode = "score";
+        myOutputValue = `Press "submit" for results`;
+      }
+    }
+    return myOutputValue;
+  }
+
+  while (gameMode == "score") {
```

Comment:
you don't have to use while here! you can use "if" instead!

Reply:
while is unnecessary here as you're changing gameMode @ line 110, so an "if" statement is sufficient!